proteus/WaveTools.pyx

```diff
@@ -1,68 +1,139 @@
-# A type of -*- python -*- file
+#!python
 #cython: embedsignature=True
 """Tools for working with water waves.
 
 The primary objective of this module is to provide solutions (exact and
 approximate) for the free surface deformation and subsurface velocity
 components of water waves. These can be used as boundary conditions, wave
 generation sources, and validation solutions for numerical wave codes.
-
-.. inheritance-diagram:: ShockCapturing
-    :parts: 2
 """
+
+
+cimport cython
 from math import pi, tanh, sqrt, exp, log, sin, cos, cosh, sinh
 import numpy as np
 import cmath as cmath
 from Profiling import logEvent
 import time as tt
 import sys as sys
 
+__all__ = ['MonochromaticWaves',
+           'RandomWaves',
+           'MultiSpectraRandomWaves',
+           'DirectionalWaves',
+           'TimeSeries',
+           'RandomWavesFast',
+           'RandomNLWaves',
+           'RandomNLWavesFast',
+           'loadExistingFunction',
+           'setVertDir',
+           'loadExistingFunction',
+           'setVertDir',
+           'setDirVector',
+           'dirCheck',
+           'reduceToIntervals',
+           'returnRectangles',
+           'returnRectangles3D',
+           'normIntegral',
+           'eta_mode',
+           'vel_mode',
+           'sigma',
+           'JONSWAP',
+           'PM_mod',
+           'cos2s',
+           'mitsuyasu',
+           'dispersion',
+           'tophat',
+           'costap',
+           'decompose_tseries']
```

Comment:
For now, the only way I can get sphinx autodoc to autogenerate module's class and function docs is to add the __all__ module data member. 
/cc @adimako @alistairbntl @tridelat @nerisaurus